File: 25주차/H-index.py
def solution(citations):
    citations.sort()
    length = len(citations)
    for i,v in enumerate(citations):
        if v >= length - i:
            return length - i
    return 0
# 프로그래머스 h-index
# n편 중 h번 이상 인용된 논문이 h편이상, 나머지 논문은 h번 이하로 인용됬을 때, h의 최댓값.
#  0   1   3   5   6
# [0] [1] [2] [3] [4] (인덱스)
# 최댓값은 h편이상이기 때문에 len(citations).
# 정렬 이후, 현재 인용된 횟수이상으로 인용된 논문 개수는 length - i(현재 인덱스).(정렬을 했으니, 자동으로 뒤는 해당 값 이상.)
# 하지만 인용 횟수 범위가 0이상이기 때문에 [0,0,0,0] 일 경우 답이 0. -> return 0으로 따로 빼준다.

File: 25주차/H-index.js
// js로 프로그래머스 풀기 - 고득점 kit 정렬.
function solution(citations) {
    var answer = -1;
    citations.sort();
    let maxV = Math.max(...citations);
    
    for (let i = maxV; i >=0 ;i--){
        let cnt = 0;
        for ( let j = citations.length-1 ; j >= 0;j--) {
            if ( i <= citations[j]) {
                cnt += 1
        }
        if ( cnt >= i) answer = Math.max(answer,i)
            
        }
    }
    return answer;
}